Sources/SwiftDocC/Infrastructure/Symbol Graph/SymbolGraphLoader.swift

```diff
@@ -232,6 +232,26 @@ struct SymbolGraphLoader {
             for (selector, _) in symbol.mixins {
                 if var symbolAvailability = (symbol.mixins[selector]?["availability"] as? SymbolGraph.Symbol.Availability) {
                     guard !symbolAvailability.availability.isEmpty else { continue }
+                    // For platforms with a fallback option (e.g., Catalyst and iOS), apply the explicit availability annotation of the fallback platform when it is not explicitly available on the primary platform.
+                    DefaultAvailability.fallbackPlatforms.forEach { (fallbackPlatform, inheritedPlatform) in
+                        guard
+                            var inheritedAvailability = symbolAvailability.availability.first(where: {
+                                $0.matches(inheritedPlatform)
+                            }),
+                            let fallbackAvailabilityIntroducedVersion = symbolAvailability.availability.first(where: {
+                                $0.matches(fallbackPlatform)
+                            })?.introducedVersion,
+                            let defaultAvailabilityIntroducedVersion = defaultAvailabilities.first(where: { $0.platformName ==  fallbackPlatform })?.introducedVersion
+                        else { return }
+                        // Ensure that the availability version is not overwritten if the symbol has an explicit availability annotation for that platform.
+                        if SymbolGraph.SemanticVersion(string: defaultAvailabilityIntroducedVersion) == fallbackAvailabilityIntroducedVersion {
```

Comment:
More of a discussion point than something to tackle in this PR, is there a reason we store `DefaultAvailability.introducedVersion` as a string rather than a semantic version type such as [`SemanticVersion`](https://github.com/apple/swift-docc/blob/main/Sources/SwiftDocC/Model/Rendering/SemanticVersion.swift#L14)? I think it would be useful to have one semantic version type which we use everywhere we reference platform build versions, similar to the discussion in https://github.com/apple/swift-docc/issues/970